src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,23 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    public static final int MIN_AGE = 35;
+    public static final int MIN_PERIOD = 10;
+    public static final String NATIONALITY = "Ukrainian";
+
+    @Override
+    public boolean test(Candidate candidate) {
+        String[] period = candidate.getPeriodsInUkr().split("-");
+        int periodResult = Integer.parseInt(period[1]) - Integer.parseInt(period[0]);
+        if (candidate.getAge() >= MIN_AGE
+                && candidate.isAllowedToVote()
+                && candidate.getNationality().equals(NATIONALITY)
+                && periodResult > MIN_PERIOD) {
+            return true;
+        }
+        return false;
```

Comment:
Could be replaced as
```suggestion
        return (candidate.getAge() >= MIN_AGE
                && candidate.isAllowedToVote()
                && candidate.getNationality().equals(NATIONALITY)
                && periodResult > MIN_PERIOD)
```